go/marshal/decode.go

```diff
@@ -217,3 +238,51 @@ func nomsValueDecoder(v types.Value, rv reflect.Value) {
 	}
 	rv.Set(reflect.ValueOf(v))
 }
+
+func sliceDecoder(t reflect.Type) decoderFunc {
+	d := decoderCache.get(t)
+	if d != nil {
+		return d
+	}
+
+	var decoder decoderFunc
+
+	d = func(v types.Value, rv reflect.Value) {
+		slice := rv.Slice(0, 0)
```

Comment:
It uses the slice that was passed in. This is for consistency with encoding/json but also to be consistent with structs (which gets filled in)